R/create_protocol.R

```diff
@@ -1,36 +1,87 @@
-#' @title Create a folder with a bookdown (R markdown) template to start a new fieldwork protocol
+#' @title Create a folder with a bookdown (R markdown) template to start a new
+#' fieldwork protocol and optionally render to html
 #'
-#' @description This function will create a new folder based on values that are passed on via the parameters and creates a R-markdown (bookdown) skeleton based on a template file to start working on a new protocol.
+#' @description This function will create a new folder based on values that are
+#' passed on via the parameters and creates a R-markdown (bookdown) skeleton
+#' based on a template file to start working on a new protocol.
+#' Optionally, the rmarkdown chapters are rendered to an html files which will
+#' be saved in a matching subfolder of the `docs` folder.
 #'
-#' @details The created folder is a new subfolder beneath `src` and contains a bookdown project. It is assumed that the src folder is a subfolder of an RStudio project with git version control.
+#'
+#' @details It is assumed that the `src` folder is a subfolder of an RStudio
+#' project with git version control.
 #'
 #' @param title A character string giving the main title of the protocol
 #' @param subtitle A character string for an optional subtitle
-#' @param short_title A character string of less than 20 characters to use in folder and filenames
-#' @param authors A character string for authors of the form First name Last name and multiple authors separated by a comma
+#' @param short_title A character string of less than 20 characters to use in
+#' folder and filenames
+#' @param authors A character vector for authors of the form First name Last
+#' name
 #' @param date A character string of the date in ISO 8601 format (YYYY-MM-DD)
-#' @param reviewers A character string for reviewers of the form First name Last name and multiple authors separated by a comma
-#' @param file_manager A character string for the name of the document maintainer of the form First name Last name
-#' @param revision A semantic version number of the form major.minor.patch. For development versions a fourth component is appended starting from .9000. The default is 0.0.0.9000 and should normally not be changed.
+#' @param reviewers A character vector for reviewers of the form First name
+#' Last name
+#' @param file_manager A character string for the name of the document
+#' maintainer of the form First name Last name
+#' @param version_number A version number of the form `YYYY.##`.
+#' For development versions `.dev` is added.
```

Comment:
> For development versions `.dev` is added.

@hansvancalster  If I'm not mistaken, this is also a step to be taken care of by the contributor, when making a new branch to work on an existing Rmd protocol. I think that is still missing from the contributing guide / workflow / diagram? Also, attention has to be paid _not_ to use the next version number (the version number to-be) in adding the `.dev` suffix, but to add the suffix behind the 'old' released number.

Reply:
@florisvdh Good point. What do we with new protocols? Give it a temporary version number like `0000.01.dev`?
Should we also add a parameter to `create_sfp` that stores the old-style revision number (e.g. 1.0) when a protocol is created from docx?

Reply:
> temporary version number

Format looks good; I'd suggest to further 'dummy' it as either `0000.00.dev` or rather even **`yyyy.00.dev`** (where `yyyy` is current year). You always have to increment the version number at release time (including possibly the year part), so `00.dev` -> `01` then makes more sense at release time.

> add a parameter to create_sfp that stores the old-style revision number

Not clear to me what the exact aim & intended results are, of what you propose. If a history of previous versions is auto-generated at rendering time, then it could make sense. Be careful with the ordering of version numbers in that case.

Reply:
OK, I will go for `yyyy.00.dev`. I guess it is probably not a problem if multiple new protocols get the same `yyyy.00.dev` number because this will be resolved at release time?

Maybe we should just break the connection with old-style revision numbers and not bother about them any more. If for some reason, someone wants to know this, it can be looked up in the docx version of the protocol from which it was converted.

Reply:
>  if multiple new protocols get the same `yyyy.00.dev` number

Each will be in a development branch, hence I don't see problems for releases and tags, as long as these YAML version numbers are effectively adjusted before merging.

> break the connection with old-style revision numbers

I would also prefer that. As the new version numbers have a different syntaxis, it will always be easy to distinguish old from new.